### ISSUE TITLE
# note: each worker gets a different seed

### ISSUE THREAD

**Issue by vgoklani** [karpathy/nanoGPT#22]
I'm confused about this line from train.py:

    torch.manual_seed(1337 + gpu_id) # note: each worker gets a different seed

Doesn't that mean each of your DDP model instances will have a different random seed?  Shouldn't the models on each DDP worker be identical, as we want to shard the data not the model.

**Comment by vgoklani** [comment]
"The DDP constructor takes a reference to the local module, and broadcasts state_dict() from the process with rank 0 to all other processes in the group to make sure that all model replicas start from the exact same state."

So the seed is irrelevant; I retract the question :)

**Comment by karpathy** [comment]
yes exactly :) ty for raising though!
